usr/src/uts/armv8/os/psci.c

```diff
@@ -120,32 +122,34 @@ psci_call(uint64_t a0, uint64_t a1, uint64_t a2, uint64_t a3)
 }
 
 void
-psci_init(void)
+psci_init(struct xboot_info *xbp)
 {
-	const struct psciinfo *pi;
+	if (psci_initialized == B_TRUE)
+		return;
 
-	pi = psciinfo_get();
-	VERIFY(pi != NULL);
+	if (xbp == NULL)
+		return;
 
-	if (pi->pi_version == PSCI_NOT_IMPLEMENTED) {
-		prom_printf("WARNING: PSCI not implemented\n");
-		psci_initialized = B_FALSE;
+	if (xbp->bi_psci_version & 0x80000000)
 		return;
-	}
 
-	switch (pi->pi_conduit) {
-	case PSCI_CONDUIT_SMC:
-		pcsi_method_is_hvc = B_FALSE;
-		break;
-	case PSCI_CONDUIT_HVC:
+	if (xbp->bi_psci_conduit == BI_PSCI_CONDUIT_HVC)
 		pcsi_method_is_hvc = B_TRUE;
-		break;
-	}
+	else if (xbp->bi_psci_conduit == BI_PSCI_CONDUIT_SMC)
+		pcsi_method_is_hvc = B_FALSE;
+	else
+		return;
```

Comment:
Do we want to `prom_printf` if we get something surprising here? I don't think we did in the loader (unlike with `psci_version`)